playbook/app/pb_kits/playbook/pb_advanced_table/SubKits/TableBody.tsx

```diff
@@ -54,93 +57,231 @@ export const TableBody = ({
+                    {/* Render custom checkbox column when we want selectableRows for non-expanding tables */}
+                    {selectableRows && !hasAnySubRows && (
+                      <td className="checkbox-cell">
+                        <Checkbox
+                            checked={row.getIsSelected()}
+                            disabled={!row.getCanSelect()}
+                            indeterminate={row.getIsSomeSelected()}
+                            name={row.id}
+                            onChange={row.getToggleSelectedHandler()}
+                        />
+                      </td>
+                    )}
+
+                    {row.getVisibleCells().map((cell: Cell<GenericObject, unknown>, i: number) => {
+                      const isPinnedLeft = columnPinning.left.includes(cell.column.id);
+                      const isLastCell = cell.column.parent?.columns?.at(-1)?.id === cell.column.id;
+
+                      return (
+                        <td
+                            align="right"
+                            className={classnames(
+                              `${cell.id}-cell position_relative`,
+                              isChrome() ? "chrome-styles" : "",
+                              isPinnedLeft && 'pinned-left',
+                              isLastCell && 'last-cell',
+                            )}
+                            key={`${cell.id}-data`}
+                        >
+                          {collapsibleTrail && i === 0 && row.depth > 0 && renderCollapsibleTrail(row.depth)}
+                          <span id={`${cell.id}-span`}>
+                            {loading ? (
+                              <LoadingCell />
+                            ) : (
+                              flexRender(cell.column.columnDef.cell, cell.getContext())
+                            )}
+                          </span>
+                        </td>
+                      );
+                    })}
+                  </tr>
+                );
+              }
+
+              if (item.type === 'loading') {
+                // Render loading indicator
+                const row = item.row;
+                const numberOfColumns = table.getAllFlatColumns().length;
+
+                return (
+                  <tr
+                      className="virtualized-table-row"
+                      key={`loading-${item.id}`}
+                      style={{
+                        ...virtualItemStyle,
+                        width: '100%',
+                      }}
+                  >
+                    <td
+                        colSpan={numberOfColumns}
+                        style={{ paddingLeft: `${row.depth === 0 ? 0.5 : (row.depth * 2)}em` }}
+                    >
+                      <LoadingInline />
+                    </td>
+                  </tr>
+                );
+              }
+
+              return null;
+            })}
+          </>
+        )}
       </tbody>
```

Comment:
@skduncan @nidaqg Another idea it would be to split this component into maybe 3??? We could have a VirtualizedTableBody, a "NormalTableBody" and the main one TableBody.